server/protocols/webadmin/webadmin-core/src/main/java/org/apache/james/webadmin/routes/HealthCheckRoutes.java

```diff
@@ -56,15 +63,17 @@ public class HealthCheckRoutes implements PublicRoutes {
 
     public static final String HEALTHCHECK = "/healthcheck";
     public static final String CHECKS = "/checks";
-    
+
     private static final String PARAM_COMPONENT_NAME = "componentName";
+    private static final String QUERY_PARAM_COMPONENT_NAMES = "checks";
 
     private final JsonTransformer jsonTransformer;
-    private final Set<HealthCheck> healthChecks;
+    private final Map<ComponentName, HealthCheck> healthCheckMap;
```

Comment:
How about keep using `Set<HealthCheck> healthChecks;`. We still can filter valid component names because the `HealthCheck` has `ComponentName` in itself. And remove the ugly and cheating reflection below.

Reply:
"We still can filter valid component names because the HealthCheck has ComponentName in itself." ----> with Set<HealthCheck> healthChecks, iterating through the set is needed and the cost is perf.

Reply:
> iterating through the set is needed and the cost is perf.

Not that much IMO:
- the set size is small
- healthcheck does not need to be super performant

Reply:
discussed with @quantranhong1999  --> ignore

Reply:
> discussed with @quantranhong1999 --> ignore

Why?

I was having same comment...

Reply:
fixed